content/en/blog/_posts/2022-XX-XX-iptables-chains.md

```diff
@@ -0,0 +1,150 @@
+outside of Kubernetes itself that make use of `KUBE-MARK-MASQ`,
+`KUBE-MARK-DROP`, or other Kubernetes-generated IPTables chains should
+start migrating away from them now.
+
+[KEP-3178]: https://github.com/kubernetes/enhancements/issues/3178
+
+## Upcoming changes
+
+Starting a few releases from now, kubelet will no longer create the
+following IPTables chains in the `nat` table:
+
+  - `KUBE-MARK-DROP`
+  - `KUBE-MARK-MASQ`
+  - `KUBE-POSTROUTING`
+
+Additionally, the `KUBE-FIREWALL` chain in the `filter` table will no
+longer have the functionality currently associated with
+`KUBE-MARK-DROP` (and it may eventually go away entirely).
+
+(This change will be phased in via the `IPTablesCleanup` feature gate,
+which will initially be off-by-default but enable-able, then later
+on-by-default but disable-able, and finally on and not disable-able.)
+
+## What to do if you use Kubernetes’s IPTables chains
+
+### If you use `KUBE-MARK-MASQ`...
+
+If you are making use of the `KUBE-MARK-MASQ` chain to cause packets
+to be masqueraded, you have two options: (1) rewrite your rules to use
+`-j MASQUERADE` directly, (2) create your own alternative “mark for
+masquerade” chain.
+
+The reason kube-proxy uses `KUBE-MARK-MASQ` is because there are lots
+of cases where it needs to call both `-j DNAT` and `-j MASQUERADE` on
+a packet, but it’s not possible to do both of those at the same time
+in IPTables; `DNAT` must be called from the `PREROUTING` (or `OUTPUT`)
+chain (because it potentially changes where the packet will be routed
+to) while `MASQUERADE` must be called from `POSTROUTING` (because the
+masqueraded source IP that it picks depends on what the final routing
+decision was).
+
+In theory, kube-proxy could have one set of rules to match packets in
+`PREROUTING`/`OUTPUT` and call `-j DNAT`, and then have a second set
+of rules to match the same packets in `POSTROUTING` and call `-j
+MASQUERADE`. But instead, for efficiency, it only matches them once,
+during `PREROUTING`/`OUTPUT`, at which point it calls `-j DNAT` and
+then calls `-j KUBE-MARK-MASQ` to set a bit on the kernel packet mark
+as a reminder to itself. Then later, during `POSTROUTING`, it has a
+single rule that matches all previously-marked packets, and calls `-j
+MASQUERADE` on them.
+
+If you have _a lot_ of rules where you need to apply both DNAT and
+masquerading to the same packets like kube-proxy does, then you may
+want a similar arrangement. But in many cases, components that use
+`KUBE-MARK-MASQ` are only doing it because they copied kube-proxy’s
+behavior without understanding why kube-proxy was doing it that way.
+Many of these components could easily be rewritten to just use
+separate DNAT and masquerade rules. (In cases where no DNAT is
+occurring then there is even less point to using `KUBE-MARK-MASQ`;
+just move your rules from `PREROUTING` to `POSTROUTING` and call `-j
+MASQUERADE` directly.)
+
+### If you use `KUBE-MARK-DROP`...
```

Comment:
I'd set an explicit fragment identifier, eg:
```suggestion
### If you use `KUBE-MARK-DROP`... {#use-case-kube-mark-drop}
```

Reply:
done, though this doesn't show up correctly in github's preview...